.github/workflows/shared-publish-to-pypi-versioned.yaml

```diff
@@ -0,0 +1,107 @@
+name: Shared Pipeline to build and publish Python packages to Pypi
+on:
+  workflow_call:
+    inputs:
+      release_type:
+        description: The type of version number to return. Must be one of [Patch, Minor or Major]
+        required: true
+        type: string
+      vulnerability_failure_severity:
+        description: The severity to fail the workflow if such vulnerability is detected. DO NOT override it unless a Jira ticket is raised. Must be one of ['CRITICAL', 'CRITICAL,HIGH' or 'CRITICAL,HIGH,MEDIUM'] (without space in between).
+        type: string
+        default: 'CRITICAL,HIGH'
+      working_dir:
```

Comment:
When do you think we would need to change this?

Reply:
When we have another python repo and the structure is a bit different (say the `pyproject.toml` file is in another folder). I think it's okay to leave it here and since it has a default value, we won't need to change it in every repo.